### DISCUSSION TITLE
Difference between query & params in getServerSideProps(context)?

### DISCUSSION THREAD

**Discussion by raajnadar** [vercel/community#477] [Help] [answered]
When I do a log of context returned from getServerSideProps, I get both query & params in the log.

What is the difference between query & params in the log?

```
query: { id: '93943aa9-73ec-4772-ba94-2dce690da2b' },
params: { id: '93943aa9-73ec-4772-ba94-2dce690da2b' },
```

**Comment by RizBizKits** [comment, marked as answer]
👋 Hey! I would say this comes down to how you design your URLs (think of the general [`query` vs. `params` use case](https://stackoverflow.com/questions/30967822/when-do-i-use-path-params-vs-query-params-in-a-restful-api)).

Whilst what is contained by `query` can be extracted from `params` in some situtations, it makes more sense to use the `query` object directly when you have a query string in your URL (e.g. `/tv/movies?genre=comedy`).

Hope I've covered what you were looking for 😄
